src/girouette/tw/border.cljc

```diff
@@ -1,10 +1,204 @@
+    :garden (fn [{:keys [component-data]}]
+              (let [{position :border-radius-position
+                     size     :border-radius-size} (into {} component-data)
+                    {side   :border-radius-side
+                     corner :border-radius-corner} (into {} [position]) ;; TODO: find more idiomatic way
+                    [size-unit val] size
+                    radius     (case size-unit
+                                 :none "0px"
+                                 :full "9999px"
+                                 (:media-query-min-width nil) (str (* 0.25 (case val
+                                                                             "sm" 0.5
+                                                                             "md" 1.5
+                                                                             "lg" 2
+                                                                             "xl" 3
+                                                                             "2xl" 4
+                                                                             1)) "rem"))
+                    directions (case side
+                                 "t" ["top-left" "top-right"]
+                                 "r" ["top-right" "bottom-right"]
+                                 "b" ["bottom-right" "bottom-left"]
+                                 "l" ["bottom-left" "top-left"]
+                                 (case corner
+                                   "tl" ["top-left"]
+                                   "tr" ["top-right"]
+                                   "br" ["bottom-right"]
+                                   "bl" ["bottom-left"]
+                                   nil))]
+                (if (nil? directions)
+                  {:border-radius radius}
+                  (into {}
+                        (map (fn [direction] [(keyword (str "border-" direction "-radius")) radius]))
+                        directions))))}
+
+   {:id     :border-width
+    :rules  "
+    border-width = <'border'> (<'-'> direction)? (<'-'> border-width-value)?
+    border-width-value = number | length | length-unit
+    "
+    :garden (fn [{:keys [component-data]}]
+              (let [{:keys [border-width-value direction]} (into {} component-data)
+                    css-prop (case direction
+                               "t" :border-top-width
+                               "r" :border-right-width
+                               "b" :border-bottom-width
+                               "l" :border-left-width
+                               :border-width)]
+                {css-prop (if (nil? border-width-value)
+                            "1px"
+                            (value-unit->css nil border-width-value {:number-unit "px"}))}))}
+   {:id     :border-color
+    :rules  "
+    border-color = <'border-'> color
+    "
+    ;; Copy-pasted from background; could be extracted into util?
```

Comment:
Yes, that's the plan. I will improve the utility functions soon.
For now, your code is good enough.